### ISSUE TITLE
Certbot no longer functioning on Fedora 25 (setuptools)

### ISSUE THREAD

**Issue by jwflory** [certbot/certbot#3931]
# Summary

When using `certbot` in the command line on Fedora 25, a stacktrace is immediately thrown and certbot becomes unusable.

# Version / OS

* **Version**: 0.9.3-1.fc25
* **Operating system**: Fedora 25 (kernel: 4.8.13-300.fc25.x86_64)

# Expected behavior

`$ certbot`

_information about certbot_

# Actual behavior

```
$ certbot

Traceback (most recent call last):
  File "/usr/bin/certbot", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2958, in <module>
    @_call_aside
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2944, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2971, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 635, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 943, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 837, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2495, in requires
    dm = self._dep_map
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2743, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2752, in _compute_dependencies
    for req in self._parsed_pkg_info.get_all('Requires-Dist') or []:
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2734, in _parsed_pkg_info
    metadata = self.get_metadata(self.PKG_INFO)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1475, in get_metadata
    return self._get(self._fn(self.egg_info, name))
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1586, in _get
    with open(path, 'rb') as stream:
IOError: [Errno 2] No such file or directory: '/usr/lib/python2.7/site-packages/setuptools-20.1.1.dist-info/METADATA'
```

**Comment by hogarthj** [comment]
Tracking this in bugzilla here: https://bugzilla.redhat.com/show_bug.cgi?id=1406191

**Comment by dashaxiong** [comment]
The latest kernel for Fedora 25 (4.8.14) doesn't have this problem. Running `sudo dnf install kernel-4.8.14` then `sudo dnf install certbot`, then `certbot --version` produces the expected output of  'certbot 0.9.3'. Running `certbot` without arguments operates normally. Possibly only an issue in 4.8.13.

**Comment by jflory7** [comment]
@dashaxiong Not sure if I am able to confirm that. I upgraded my kernel and followed your steps, and had the same output.

```
$ uname -a
Linux tronstudentritedu 4.8.14-300.fc25.x86_64 #1 SMP Mon Dec 12 16:31:04 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
$ certbot
Traceback (most recent call last):
  File "/usr/bin/certbot", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2958, in <module>
    @_call_aside
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2944, in _call_aside
    f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2971, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 635, in _build_master
    ws.require(__requires__)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 943, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 837, in resolve
    new_requirements = dist.requires(req.extras)[::-1]
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2495, in requires
    dm = self._dep_map
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2743, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2752, in _compute_dependencies
    for req in self._parsed_pkg_info.get_all('Requires-Dist') or []:
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2734, in _parsed_pkg_info
    metadata = self.get_metadata(self.PKG_INFO)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1475, in get_metadata
    return self._get(self._fn(self.egg_info, name))
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1586, in _get
    with open(path, 'rb') as stream:
IOError: [Errno 2] No such file or directory: '/usr/lib/python2.7/site-packages/setuptools-20.1.1.dist-info/METADATA'
```

**Comment by dashaxiong** [comment]
@jflory7 I'm sorry, I can't replicate the issue yet. I have tried with a fresh install of Fedora 25 in a VM as I usually use Kubuntu. One thing I note in the error output is that `pkg_resources` is looking for `setuptools-20.1.1.dist-info` and failing to find that. On a fresh install of Fedora 25, after installing certbot with `dnf`, the setuptools site-package is at '25.1.1'. Could you provide the output of `find /usr/lib/python2.7 -type d -regextype sed -iregex ".*/setuptools.*dist.*"`?

**Comment by dashaxiong** [comment]
@jflory7 Thanks for the info. I've been able to replicate the problem here.

Since you have setuptools 25.1.1 installed via dnf, the 20.1.1 directory is redundant. It's mere presence there fools setuptools into thinking that it must be installed, when in fact it appears not to be. I was able to replicate the problem by running:

```
mkdir /usr/lib/python2.7/site-packages/setuptools-20.1.1.dist-info
```

After that `certbot --version` gives the precise error output reported above.

Try running `rm -rf /usr/lib/python2.7/site-packages/setuptools-20.1.1.dist-info/`

Then run `certbot --version` to check.
